### ISSUE TITLE
Programatic bug report templates

### ISSUE THREAD

**Issue by sambostock** [rubocop/rubocop#14280]
## Is your feature request related to a problem? Please describe.

It is useful for both maintainers and consumers for bug reports to be reproducible.

Currently, our bug report template includes the following:

- Expected behavior
- Actual behavior
- Steps to reproduce the problem
- RuboCop version

However, it would be nice if we could provide consumers a way to package up a reproduction test case.

## Describe the solution you'd like

Rails has [bug report templates](https://github.com/rails/rails/tree/main/guides/bug_report_templates) which consumers can use to reproduce bug they're running into in an isolated environment, and which maintainers can use to reproduce the issue on their end. For example, the `bug_report_templates/generic.rb`

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  gem "rails"
  # If you want to test against edge Rails replace the previous line with this:
  # gem "rails", github: "rails/rails", branch: "main"
end

require "active_support"
require "active_support/core_ext/object/blank"
require "minitest/autorun"

class BugTest < ActiveSupport::TestCase
  def test_stuff
    assert "zomg".present?
    assert_not "".present?
  end
end
```

(There are also other more specific templates for minimally reproducing issues in various Rails frameworks, like Active Record or Active Job.)

**It would be nice if Rubocop provided something similar.**

For example, we could provide two templates:

- one template for reporting bugs specific to a single cop (modelled after our per-cop-class specs)
- one template for reporting bugs in the CLI or involving interactions between multiple cops (modelled after the CLI specs)

## Describe alternatives you've considered

Status quo: consumers continue to provide free-form reproduction steps.

## Additional context

A side benefit of these templates is that they provide an isolated environment in which to experiment with features. For example, I have used [`bug_report_templates/active_record.rb`](https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_record.rb) before to play around with a particular method before to gain an understanding of how it behaves, or confirm I'm using it correctly.

These can also be useful for answering the question: "is a given behaviour specific to my app?"

**Comment by koic** [comment]
RuboCop doesn’t have the complex library or middleware dependencies that Rails does. The current template is simple and already contains all the information maintainers need. Thank you.
